src/julia_to_gap.jl

```diff
@@ -46,44 +46,52 @@ The following `julia_to_gap` conversions are supported by GAP.jl.
 | `UnitRange{T}`, `StepRange{T, S}`    | `IsRange`    |
 | `Function`                           | `IsFunction` |
 """
-julia_to_gap(x::FFE) = x    # Default for actual GAP objects is to do nothing
-julia_to_gap(x::Bool) = x   # Default for actual GAP objects is to do nothing
+function julia_to_gap end
+
+# default
+julia_to_gap(x, cache::GapCacheDict = nothing; recursive::Bool = false) = julia_to_gap_internal(x, cache, recursive)
```

Comment:
Isn't this in fact the only remaining method for `julia_to_gap`? Then perhaps we can just do this:
```suggestion
julia_to_gap(x, cache::GapCacheDict = nothing; recursive::Bool = false) = julia_to_gap_internal(x, cache, recursive)
```

Reply:
Note that `src/constructors.jl` has this:
```julia
## Handle "conversion" to GAP.Obj and GAP.GapObj (may occur in recursions).
Obj(x::Obj) = x
GapObj(x::GapObj) = x

## Handle conversion of Julia objects to GAP objects
Obj(obj; recursive::Bool = false) = julia_to_gap(obj, IdDict(); recursive)::Obj
GapObj(obj; recursive::Bool = false) = julia_to_gap(obj, IdDict(); recursive)::GapObj

Obj(obj, recursive::Bool) = julia_to_gap(obj, IdDict(); recursive)::Obj
GapObj(obj, recursive::Bool) = julia_to_gap(obj, IdDict(); recursive)::GapObj
```

These are now suboptimal because we create those `IdDict` unconditionally. I assume those could / should all be replaced by `nothing` now?

Alternatively one could rename the last `julia_to_gap` method here to `GapObj`, adjust `src/constructors.jl` accordingly, and either just drop `julia_to_gap` or turn it into an alias resp. mark it as deprecated.

At that point one could also consider renaming `julia_to_gap_internal` to `GapObj_internal`.

Reply:
If we want to get rid of `julia_to_gap`, and recommend `GapObj` instead, also the documentation has to be adjusted.
The only logical problem is that then `GapObj` can return also objects of type `GAP.Obj` that are not of type `GapObj`. (We had talked about this.)
I will leave this renaming to a separate commit, and push the other changes now.